pkg/bgp/manager/subscriber.go

```diff
@@ -153,6 +153,9 @@ func (m *Manager) beforeRun() {
 		for _, k := range retryKeys {
 			l.WithField("service", k).Warn("over 60 retries, giving up reconciliation")
 		}
+		// This is a fatal error as the BGP manager must synchronize existing services before proceeding.
+		// We panic here to crash the operator and allow for a restart.
+		panic("failed to reconcile some services on BGP manager startup")
```

Comment:
Using panic wouldn’t allow other goroutines to shut down gracefully, so I think we should do in a different way.

How about using `hive.Shutdowner` instead?
I think we could make it work by adding `hive.Shutdowner` to `registerLegacyOnLeader()` since hive would inject it automatically when the cell is called.

Reply:
Thanks, I fixed to use `Shutdowner`.